crates/red_knot_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -133,6 +133,8 @@ pub(crate) struct SearchPaths {
     /// in terms of module-resolution priority until we've discovered the editable installs
     /// for the first `site-packages` path
     site_packages: Vec<SearchPath>,
+
+    typeshed_versions: Cow<'static, TypeshedVersions>,
```

Comment:
An alternative to storing `typeshed_versions` on `SearchPaths` would be to store the `TypeshedVersions` alongside the `SearchPath` (for the Custom or vendored stdlib variants). I don't really have an opinion on what's better. Let me know if you have  ;)